tests/verifyAnnotations_test.js

```diff
@@ -0,0 +1,127 @@
+const { pmmInventoryPage, dashboardPage } = inject();
+const assert = require('assert');
+
+const annotation = new DataTable(['annotationName', 'service', 'dashboard']);
+
+annotation.add(['annotation-for-mongo', pmmInventoryPage.fields.mongoServiceName, dashboardPage.mongoDbInstanceSummaryDashboard.url]);
+annotation.add(['annotation-for-postgres', pmmInventoryPage.fields.pdphsqlServiceName, dashboardPage.postgresqlInstanceSummaryDashboard.url]);
+annotation.add(['annotation-for-mysql', pmmInventoryPage.fields.mysqlServiceName, dashboardPage.mysqlInstanceSummaryDashboard.url]);
+
+Feature('Test annotation on dashboards');
+
+Before(async ({ I }) => {
+  await I.Authorize();
+});
+
+Scenario(
+  'PMM-T878 - Verify adding annotation for service name: pmm-server-postgresql and node name: pmm-server @nightly @dashboards',
+  async ({ I, dashboardPage, annotationAPI }) => {
+    const postgresAnnotation = 'annotation-for-postgres';
+
+    await annotationAPI.setAnnotation(postgresAnnotation, 'PMM-T878', 'pmm-server', 'pmm-server-postgresql');
+    I.amOnPage(`${dashboardPage.postgresqlInstanceSummaryDashboard.url}`);
+    dashboardPage.waitForDashboardOpened();
+    await dashboardPage.applyFilter('Service Name', 'pmm-server-postgres');
+    dashboardPage.verifyAnnotationsLoaded(postgresAnnotation, 3);
+    I.seeElement(dashboardPage.annotationText(postgresAnnotation), 10);
+  },
+);
+
+Data(annotation).Scenario(
+  'PMM-T878 - Verify adding annotation specific dashboard @nightly @dashboards',
+  async ({
+    I, dashboardPage, pmmInventoryPage, annotationAPI, inventoryAPI, current,
+  }) => {
+    const { annotationName } = current;
+
+    I.amOnPage(pmmInventoryPage.url);
+    I.waitForVisible(current.service, 10);
+    const serviceName = await I.grabTextFrom(current.service);
+    const nodeID = await pmmInventoryPage.getNodeId(serviceName);
+    const nodeName = await inventoryAPI.getNodeName(nodeID);
+
+    await annotationAPI.setAnnotation(annotationName, 'PMM-T878', nodeName, serviceName);
+    I.amOnPage(current.dashboard);
+    dashboardPage.waitForDashboardOpened();
+    await dashboardPage.applyFilter('Service Name', serviceName);
+    if (annotationName === 'annotation-for-postgres') {
+      dashboardPage.verifyAnnotationsLoaded(annotationName, 3);
+    } else {
+      dashboardPage.verifyAnnotationsLoaded(annotationName, 1);
+    }
+
+    I.seeElement(dashboardPage.annotationText(annotationName), 10);
+  },
+);
+
+Data(annotation).Scenario(
```

Comment:
I think there is no need to try different service names. Better to combine these 3 tests (wrong service name, wrong node name, without service name)